backend/clin_overview/models.py

```diff
@@ -0,0 +1,204 @@
+from django.db import models
+from django.utils.translation import gettext as _
+from django.core.validators import MaxValueValidator, MinValueValidator
+from django.db.models import F, Q
+
+
+# Enums for model
+class CudStageFIGO2014(models.TextChoices):
+    IA = "IA"
+    IB = "IB"
+    IC1 = "IC1"
+    IC2 = "IC2"
+    IC3 = "IC3"
+    IIA = "IIA"
+    IIB = "IIB"
+    IIIA1i = "IIIA1i"
+    IIIA1ii = "IIIA1ii"
+    IIIA2 = "IIIA2"
+    IIIB = "IIIB"
+    IIIC = "IIIC"
+    IVA = "IVA"
+    IVB = "IVB"
+
+
+class CudTreatmentPhase(models.TextChoices):
+    PROGRESSION = "PROGRESSION", _("Progression")
+    FOLLOWUP = "FOLLOWUP", _("Follow.-up")
+    PRIMARYCHEMO = "PRIMARYCHEMO", _("Primary chemotherapy")
+    DRUGTRIAL = "DRUGTRIAL", _("Drug trial")
+    HORMONAL = "HORMONAL", _("Hormonal treatment")
+
+
+class CudTreatmentStrategy(models.TextChoices):
+    NACT = "NACT", _("Neo-adjuvant chemotherapy")
+    PDS = "PDS", _("Primary debulking surgery")
+
+
+class TissueType(models.TextChoices):
+    OVARY = "OVARY", _("Ovary")
+    OMENTUM = "OMENTUM", _("Omentum")
+    PERITONEUM = "PERITONEUM", _("Peritoneum")
+    LYMPHNODE = "LYMPHNODE", _("lymph node")
+    MESOTHELIUM = "MESOTHELIUM", _("Mesothelium")
+    ASCITES = "ASCITES", _("Ascites")
+
+
+class CudHistology(models.TextChoices):
+    HGSOC = "HGSOC", _("High grade serous ovarian cancer")
+    MUCINOUS = "MUCINOUS", _("Mucinous ovarian cancer")
+    ENDOMETRIOID = "ENDOMETRIOID", _("Endometrioid")
+
+
+class CudSurvival(models.TextChoices):
+    DEATH_UNKNOWN = "DEATHUNKNOWN", _("Death due to unknown reason")
+    DEATH_CANCER = "DEATHCANCER", _("Death due to cancer")
+    ALIVE = "ALIVE", _("Alive")
+
+
+class CudPrimaryTherapyOutcome(models.TextChoices):
+    PARTIAL = "PARTIAL", _("Partial response")
+    COMPLETE = "COMPLETE", _("Complete response")
+    PROGRESSIVE = "PROGRESSIVE", _("Progressive disease")
+    DEATH = "DEATH", _("Death during therapy")
+    STOPPED = "STOPPED", _("Stopped")
+    STOPPED_SIDEFFECTS = "SIDEEFFECTSTOP", _("Stopped due to side effects")
+    UNKNOWN = "UNKNOWN", _("Unknown response")
+
+
+class MainetenaceTherapy(models.TextChoices):
+    NOMAINTENANCE = "NOMAINTENANCE", _("No maintenance therapy")
+    BEVACIZUMAB = "BEVACIZUMAB", _("Bevacizumab")
+    PARPI = "PARPI", _("PARP inhibition therapy")
+
+
+class ClinicalData(models.Model):
+    patient = models.CharField(max_length=20, unique=True)
+    extra_patient_info = models.CharField(max_length=400, blank=True)
```

Comment:
Unlimited text fields can be a security issue: what if a malicious user upload gigabytes of text? We don't want to make anything crash and open a breach. I don't know the current state of security management in the frameworks we use, but a few years ago it was a classical concern. Maybe we could use the database's max field limit, which should be large enough?

Reply:
I cannot see how an unlimited text field in a database could cause a security issue and what specific maximum length would prevent such an issue. Maybe a denial of service is possible, but even in these cases, such gigabyte-strings should be blocked before they even reach the database.

Afaik, we are not going to provide users with text fields for entering such data. Instead, we will have a mechanism that imports data from an external source. In my opinion, that would be a proper place to block ridiculously long strings ... if needed.

Reply:
I double checked and indeed, it's not on the [OWASP's cheatsheet about input validation](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html), so I guess you're right (and I'm too old).

Reply:
Django also enforces the `max_length` argument to the `CharField` attributes and it can't be set to `None`. This is the error message if it is not provided: `(fields.E120) CharFields must define a 'max_length' attribute.`